core/src/test/java/com/gentics/mesh/core/schema/SchemaAllowNodeFieldTest.java

```diff
@@ -0,0 +1,108 @@
+import com.gentics.mesh.core.rest.schema.FieldSchema;
+import com.gentics.mesh.core.rest.schema.impl.ListFieldSchemaImpl;
+import com.gentics.mesh.core.rest.schema.impl.NodeFieldSchemaImpl;
+import com.gentics.mesh.core.rest.schema.impl.SchemaCreateRequest;
+import com.gentics.mesh.core.rest.schema.impl.SchemaResponse;
+import com.gentics.mesh.test.context.AbstractMeshTest;
+import com.gentics.mesh.test.context.MeshTestSetting;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Collections;
+
+import static com.gentics.mesh.test.ClientHelper.call;
+import static com.gentics.mesh.test.TestDataProvider.PROJECT_NAME;
+import static com.gentics.mesh.test.TestSize.FULL;
+import static com.gentics.mesh.test.context.ElasticsearchTestMode.TRACKING;
+import static io.netty.handler.codec.http.HttpResponseStatus.BAD_REQUEST;
+
+@MeshTestSetting(elasticsearch = TRACKING, testSize = FULL, startServer = true)
+public class SchemaAllowNodeFieldTest extends AbstractMeshTest {
+    private String nodeUuid;
+
+    @Before
+    public void setUp() throws Exception {
+        nodeUuid = tx(() -> folder("2015").getUuid());
+    }
+
+    private void createSchema(FieldSchema field) {
+        field.setName("testField");
+
+        SchemaCreateRequest req = new SchemaCreateRequest();
+        req.setName("test");
+        req.setFields(Collections.singletonList(field));
+
+        SchemaResponse response = client().createSchema(req).blockingGet();
+        client().assignSchemaToProject(PROJECT_NAME, response.getUuid()).blockingAwait();
+    }
+
+    private void createNode(Field field) {
+        NodeCreateRequest req = new NodeCreateRequest();
+        req.setLanguage("en");
+        req.setSchemaName("test");
+        req.setParentNodeUuid(nodeUuid);
+        FieldMapImpl fieldMap = new FieldMapImpl();
+        fieldMap.put("testField", field);
+        req.setFields(fieldMap);
+
+        client().createNode(PROJECT_NAME, req).blockingAwait();
+    }
+    
+    private void createNodeAndExpectFailure(Field field) {
+        NodeCreateRequest req = new NodeCreateRequest();
+        req.setLanguage("en");
+        req.setSchemaName("test");
+        req.setParentNodeUuid(nodeUuid);
+        FieldMapImpl fieldMap = new FieldMapImpl();
+        fieldMap.put("testField", field);
+        req.setFields(fieldMap);
+
+        call(() -> client().createNode(PROJECT_NAME, req), BAD_REQUEST,"node_error_invalid_schema_field_value","testField","test");
+    }
+
+    private void runTest(FieldSchema schemaField, Field nodeField) {
+        createSchema(schemaField);
+        createNode(nodeField);
+    }
+
+    private void runTestAndExpectFailure(FieldSchema schemaField, Field nodeField) {
+        createSchema(schemaField);
+        createNodeAndExpectFailure(nodeField);
+    }
+
+    @Test
+    public void node() {
+        runTest(
+                new NodeFieldSchemaImpl().setAllowedSchemas("test"),
+                new NodeFieldImpl().setUuid(nodeUuid));
```

Comment:
The referenced node is a folder. `nodeUuid` is set in the `setup()` method, where you can see that this is a folder. You need to create a node of the `test` schema and use that UUID here. Same for the other tests in this class.

Reply:
Thanks @philippguertler , I have updated it.